### DISCUSSION TITLE
Pytest vs unittest?

### DISCUSSION THREAD

**Discussion by IvanaH8** [NIVANorge/lake_liming_app#18] [Q&A] [answered]
@nmarquesin @JamesSample what would you prefer to implement for testing? I have only used pytest so far but happy to look into unittest if that is preferred. Thanks :)

**Comment by JamesSample** [comment, marked as answer]
PyTest is fine by me :-)

**Comment by nmarquesin** [comment]
I had only used PyTest before this week but now had a look at unittest and is also pretty ok to implement.

But regardless of which of the two libraries we choose, I am running into issues when implementing **automated browser testing** because of [Streamlit current limitations](https://github.com/streamlit/streamlit/issues/3888). 

I've tried using [selenium](https://selenium-python.readthedocs.io/navigating.html) by itself and [with robot framework](https://robotframework.org/SeleniumLibrary/SeleniumLibrary.html#Locating%20elements), because I saw [someone suggesting it for Streamlit testing](https://blog.devgenius.io/testing-streamlit-a1f1fd48ce8f). The problem I'm stuck with is that I can't find a way to select the side bar generated by Streamlit, and therefore I cannot navigate the app programmatically to test its functionality.

Are we interested in running automated tests as well as unit tests? If so, would an "upgrade" to a django app be completely out of question for the final delivery?
